include/ocpp/v201/charge_point.hpp

```diff
@@ -697,12 +705,38 @@ class ChargePoint : public ChargePointInterface, private ocpp::ChargingStationBa
     void handle_data_transfer_req(Call<DataTransferRequest> call);
 
     // general message handling
-    template <class T> bool send(ocpp::Call<T> call) {
-        this->message_queue->push(call);
-        return true;
+    template <class T> bool send(ocpp::Call<T> call, const bool initiated_by_trigger_message = false) {
```

Comment:
Can we move the implementation to the cpp file? Only the charge_point needs to call these functions so we don't need to expose the implementation.

Reply:
Yes, this should be possible as this function is only called from within the `charge_point.cpp`.  Having a wrapper class for ocpp::Call<T> which uses type erasure (like `std::variant`), most of the template usage could be removed here.

Reply:
I don't necessary mean that the whole template function needs to be removed. More that template implementation does not have to reside in the header file. You can perfectly create the template prototype here and then define the template implementation in the CPP file as long as all the methods that use it have the actual implementation before they use the templated call.